src/isomorphic/devtools/ReactPerfAnalysis.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactPerfAnalysis
+ */
+
+'use strict';
+
+var ReactPerfDevtool = require('ReactPerfDevtool');
+var ReactPerfTimers = require('ReactPerfTimers');
+
+var {start, stop, getHistory} = ReactPerfDevtool;
+
+function getExclusiveStats(history = getHistory()) {
+  var statsByComponent = {};
+  var affectedIDsByDisplayName = {};
+
+  history.forEach(entry => {
+    var {measurements, instanceTree} = entry;
+
+    measurements.forEach(measurement => {
+      var {duration, id, type} = measurement;
+      var displayName = instanceTree[id].displayNames.self;
+
+      var stats = statsByComponent[displayName];
+      if (!stats) {
+        affectedIDsByDisplayName[displayName] = {};
+        stats = statsByComponent[displayName] = {
+          displayName,
+          instanceCount: 0,
+          renderCount: 0,
+          renderDuration: 0,
+          mountCount: 0,
+          mountDuration: 0,
+          updateCount: 0,
+          updateDuration: 0,
+          unmountCount: 0,
+          unmountDuration: 0,
+          totalDuration: 0,
+        };
+      }
+
+      affectedIDsByDisplayName[displayName][id] = true;
+      stats.totalDuration += duration;
+      stats.instanceCount = Object.keys(
+        affectedIDsByDisplayName[displayName]
+      ).length;
+
+      switch (type) {
+        case ReactPerfTimers.COMPOSITE_CONSTRUCTOR:
+        case ReactPerfTimers.COMPOSITE_WILL_MOUNT:
+        case ReactPerfTimers.COMPOSITE_DID_MOUNT:
+        case ReactPerfTimers.NATIVE_MOUNT:
+          stats.mountCount++;
+          stats.mountDuration += duration;
+          break;
+        case ReactPerfTimers.ATTACH_REFS:
+        case ReactPerfTimers.DETACH_REFS:
+        case ReactPerfTimers.COMPOSITE_WILL_RECEIVE_PROPS:
+        case ReactPerfTimers.COMPOSITE_SHOULD_UPDATE:
+        case ReactPerfTimers.COMPOSITE_WILL_UPDATE:
+        case ReactPerfTimers.COMPOSITE_DID_UPDATE:
+        case ReactPerfTimers.NATIVE_UPDATE:
+          stats.updateCount++;
+          stats.updateDuration += duration;
+          break;
+        case ReactPerfTimers.COMPOSITE_RENDER:
+          stats.renderCount++;
+          stats.renderDuration += duration;
+          break;
+        case ReactPerfTimers.COMPOSITE_WILL_UNMOUNT:
+        case ReactPerfTimers.NATIVE_UNMOUNT:
+          stats.unmountCount++;
+          stats.unmountDuration += duration;
+          break;
+      }
```

Comment:
I don’t think it really matters here. This is temporary code and the style is up to React team. It will probably change 20 times before this is ready to merge :-)